lib/posix/pthread.c

```diff
@@ -198,38 +198,59 @@ void __z_pthread_cleanup_pop(int execute)
 	}
 }
 
-static bool is_posix_policy_prio_valid(uint32_t priority, int policy)
```

Comment:
I don't believe this function needs to change at all.  Please revert. Posix priorities are (by definition) positive, since negative priorities would interfere with a return value of -1 signifying error.